### PR TITLE
Add reader, ParallelExecutor and refine for Transformer

### PR DESCRIPTION
**Pull request by guoshengCS**
Add reader, ParallelExecutor and refine for Transformer

### CODE REVIEW

File: fluid/neural_machine_translation/transformer/config.py [hunk 1]
```diff
@@ -1,43 +1,50 @@
...
-    pass_num = 2
-
+    pass_num = 30
     # the number of sequences contained in a mini-batch.
-    batch_size = 64
-
+    batch_size = 32
     # the hyper parameters for Adam optimizer.
-    learning_rate = 0.001
+    # This static learning_rate will multiply LearningRateScheduler
```
Comment (pkuyym) [inline fluid/neural_machine_translation/transformer/config.py]: will multiply --> will be multiplied to

File: fluid/neural_machine_translation/transformer/reader.py [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+            try:
+                sample = self._sample_generator.next()
+                self._pool.append(sample)
+            except StopIteration as e:
+                self._end = True
+                break
+
+        if self._sort:
+            self._pool.sort(
+                key=lambda sample: max(len(sample[0]), len(sample[1])) if len(sample) > 1 else len(sample[0])
```
Comment (pkuyym) [inline fluid/neural_machine_translation/transformer/reader.py]: Exceed 80 columns.

File: fluid/neural_machine_translation/transformer/reader.py [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+                 max_length=100,
+                 shuffle=True,
+                 shuffle_batch=False,
+                 use_token_batch=False,
+                 delimiter="\t",
+                 start_mark="<s>",
+                 end_mark="<e>",
+                 unk_mark="<unk>",
+                 seed=0):
+        """
```
Comment (pkuyym) [inline fluid/neural_machine_translation/transformer/reader.py]: The comments shouldn't be here. 

```
class DataReader(object):
"""Comments for the constructor"""
```

File: fluid/neural_machine_translation/transformer/reader.py [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+        Load all data from files and set the settings to make mini-batches.
+
+        :param src_vocab_fpath: The path of vocabulary file of source language.
+        :type src_vocab_fpath: basestring
+        :param trg_vocab_fpath: The path of vocabulary file of target language.
+        :type trg_vocab_fpath: basestring
+        :param fpattern: The pattern to match data files.
```
Comment (pkuyym) [inline fluid/neural_machine_translation/transformer/reader.py]: better to give an example.

File: fluid/neural_machine_translation/transformer/reader.py [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+        :type fpattern: basestring
+        :param batch_size: The number of sequences contained in a mini-batch.
+            or the maximum number of tokens (include paddings) contained in a
+            mini-batch.
+        :type batch_size: int
+        :param pool_size: The buffer size to pool data.
```
Comment (pkuyym) [inline fluid/neural_machine_translation/transformer/reader.py]: Size of pool buffer.

File: fluid/neural_machine_translation/transformer/reader.py [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+        :type pool_size: int
+        :param sort_type: The grain to sort by length: 'global' for all
+            instances; 'pool' for instances in pool; 'none' for no sort.
+        :type sort_type: basestring
+        :param sort_type: The grain to sort by length: 'global' for all
```
Comment (pkuyym) [inline fluid/neural_machine_translation/transformer/reader.py]: duplicated.

**Review by pkuyym — CHANGES_REQUESTED** [review]

File: fluid/neural_machine_translation/transformer/config.py (hunk 1 above)
Comment (guoshengCS) [reply fluid/neural_machine_translation/transformer/config.py]: Done.

File: fluid/neural_machine_translation/transformer/reader.py (hunk 2 above)
Comment (guoshengCS) [reply fluid/neural_machine_translation/transformer/reader.py]: Done.

File: fluid/neural_machine_translation/transformer/reader.py (hunk 6 above)
Comment (guoshengCS) [reply fluid/neural_machine_translation/transformer/reader.py]: Done.

File: fluid/neural_machine_translation/transformer/reader.py (hunk 5 above)
Comment (guoshengCS) [reply fluid/neural_machine_translation/transformer/reader.py]: Done.

File: fluid/neural_machine_translation/transformer/reader.py (hunk 4 above)
Comment (guoshengCS) [reply fluid/neural_machine_translation/transformer/reader.py]: Done.

**Review by pkuyym — APPROVED** [review]
LGTM

**Pull request merged**